src/controller/CHIPDeviceController.cpp

```diff
@@ -1209,192 +1143,167 @@ CHIP_ERROR DeviceCommissioner::ValidateAttestationInfo(const ByteSpan & attestat
-
-    CommissioneeDeviceProxy * device = mDeviceBeingCommissioned;
-
-    ChipLogProgress(Controller, "Getting certificate chain for the device from the issuer");
-
-    DeviceAttestationVerifier * dacVerifier = GetDeviceAttestationVerifier();
-
-    P256PublicKey dacPubkey;
-    ReturnErrorOnFailure(ExtractPubkeyFromX509Cert(device->GetDAC(), dacPubkey));
-
-    // Retrieve attestation challenge
-    ByteSpan attestationChallenge = mSystemState->SessionMgr()
-                                        ->GetSecureSession(device->GetSecureSession().Value())
-                                        ->GetCryptoContext()
-                                        .GetAttestationChallenge();
-
-    // The operational CA should also verify this on its end during NOC generation, if end-to-end attestation is desired.
-    ReturnErrorOnFailure(dacVerifier->VerifyNodeOperationalCSRInformation(NOCSRElements, attestationChallenge, AttestationSignature,
-                                                                          dacPubkey, device->GetCSRNonce()));
-
-    mOperationalCredentialsDelegate->SetNodeIdForNextNOCRequest(device->GetDeviceId());
-
-    FabricInfo * fabric = mSystemState->Fabrics()->FindFabricWithIndex(mFabricIndex);
-    mOperationalCredentialsDelegate->SetFabricIdForNextNOCRequest(fabric->GetFabricId());
+    OperationalCredentials::Commands::AddTrustedRootCertificate::Type request;
+    request.rootCertificate = rcac;
+    return SendCommand<OperationalCredentialsCluster>(device, request, OnAddTrustedRootCertificate);
+}
 
-    return mOperationalCredentialsDelegate->GenerateNOCChain(NOCSRElements, AttestationSignature, device->GetDAC(), ByteSpan(),
-                                                             ByteSpan(), &mDeviceNOCChainCallback);
+TRAMPOLINE(AddTrustedRootCertificate, const app::DataModel::NullObjectType &);
+void DeviceCommissioner::OnAddTrustedRootCertificate(const app::DataModel::NullObjectType & data)
+{
+    StopSessionIf(mState != State::Initialized, CHIP_ERROR_INCORRECT_STATE);
+    StopSessionIf(mDeviceBeingCommissioned == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    StopSessionIf(SendOperationalCertificate(mDeviceBeingCommissioned, mDeviceBeingCommissioned->GetNOCCert(),
+                                             mDeviceBeingCommissioned->GetICACert()));
 }
 
 CHIP_ERROR DeviceCommissioner::SendOperationalCertificate(CommissioneeDeviceProxy * device, const ByteSpan & nocCertBuf,
                                                           const ByteSpan & icaCertBuf)
 {
-    VerifyOrReturnError(device != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    chip::Controller::OperationalCredentialsCluster cluster;
-    cluster.Associate(device, 0);
+    OperationalCredentials::Commands::AddNOC::Type request;
+    request.NOCValue      = mDeviceBeingCommissioned->GetNOCCert();
+    request.ICACValue     = MakeOptional(mDeviceBeingCommissioned->GetICACert());
```

Comment:
This used to send `nocCertBuf` and `icaCertBuf`.  Why the change to not use the arguments?

Reply:
I have reverted that.